openhands/server/session/agent_session.py

```diff
@@ -197,14 +196,24 @@ async def start(
         finally:
             self._starting = False
             success = finished and runtime_connected
-            self.logger.info(
-                'Agent session start',
-                extra={
-                    'signal': 'agent_session_start',
-                    'success': success,
-                    'duration': (time.time() - started_at),
-                },
-            )
+            duration = (time.time() - started_at)
+
+            log_metadata = {
+                'signal': 'agent_session_start',
+                'success': success,
+                'duration': duration,
+                'has_replay': replay_json is not None
+            }
+            if success:
+                self.logger.info(
+                    f'Agent session start succeeded in {duration}s',
+                    extra=log_metadata
+                )
+            else:
+                self.logger.error(
+                    f'Agent session start failed in {duration}s',
+                    extra=log_metadata
```

Comment:
Sometimes this indicates an exception, which is logged in the parent call session::start().

Why are AgentSession and Session different classes again?

Reply:
They don't really need to be different classes--one just holds the other as a member

Reply:
Also good call splitting these logs--this tricked us the other day